File: sdkwork-rtc-sdk-go/rtcstandard/provider_package_catalog.go
package rtcstandard

type RtcProviderPackageCatalogEntry struct {
    ProviderKey         string
    PluginId            string
    DriverId            string
    PackageIdentity     string
    ManifestPath        string
    ReadmePath          string
    SourcePath          string
    SourceSymbol        string
    Builtin             bool
    RootPublic          bool
    Status              string
    RuntimeBridgeStatus string
}

type RtcProviderPackageCatalog struct{}

var OFFICIAL_RTC_PROVIDER_PACKAGES = []RtcProviderPackageCatalogEntry{
    {ProviderKey: "volcengine", PluginId: "rtc-volcengine", DriverId: "sdkwork-rtc-driver-volcengine", PackageIdentity: "github.com/sdkwork/rtc-sdk-provider-volcengine", ManifestPath: "providers/rtc-sdk-provider-volcengine/go.mod", ReadmePath: "providers/rtc-sdk-provider-volcengine/README.md", SourcePath: "providers/rtc-sdk-provider-volcengine/provider_package_contract.go", SourceSymbol: "RtcProviderVolcenginePackageContract", Builtin: true, RootPublic: false, Status: "future-runtime-bridge-only", RuntimeBridgeStatus: "reserved"},
    {ProviderKey: "aliyun", PluginId: "rtc-aliyun", DriverId: "sdkwork-rtc-driver-aliyun", PackageIdentity: "github.com/sdkwork/rtc-sdk-provider-aliyun", ManifestPath: "providers/rtc-sdk-provider-aliyun/go.mod", ReadmePath: "providers/rtc-sdk-provider-aliyun/README.md", SourcePath: "providers/rtc-sdk-provider-aliyun/provider_package_contract.go", SourceSymbol: "RtcProviderAliyunPackageContract", Builtin: true, RootPublic: false, Status: "future-runtime-bridge-only", RuntimeBridgeStatus: "reserved"},
    {ProviderKey: "tencent", PluginId: "rtc-tencent", DriverId: "sdkwork-rtc-driver-tencent", PackageIdentity: "github.com/sdkwork/rtc-sdk-provider-tencent", ManifestPath: "providers/rtc-sdk-provider-tencent/go.mod", ReadmePath: "providers/rtc-sdk-provider-tencent/README.md", SourcePath: "providers/rtc-sdk-provider-tencent/provider_package_contract.go", SourceSymbol: "RtcProviderTencentPackageContract", Builtin: true, RootPublic: false, Status: "future-runtime-bridge-only", RuntimeBridgeStatus: "reserved"},
    {ProviderKey: "agora", PluginId: "rtc-agora", DriverId: "sdkwork-rtc-driver-agora", PackageIdentity: "github.com/sdkwork/rtc-sdk-provider-agora", ManifestPath: "providers/rtc-sdk-provider-agora/go.mod", ReadmePath: "providers/rtc-sdk-provider-agora/README.md", SourcePath: "providers/rtc-sdk-provider-agora/provider_package_contract.go", SourceSymbol: "RtcProviderAgoraPackageContract", Builtin: false, RootPublic: false, Status: "future-runtime-bridge-only", RuntimeBridgeStatus: "reserved"},
    {ProviderKey: "zego", PluginId: "rtc-zego", DriverId: "sdkwork-rtc-driver-zego", PackageIdentity: "github.com/sdkwork/rtc-sdk-provider-zego", ManifestPath: "providers/rtc-sdk-provider-zego/go.mod", ReadmePath: "providers/rtc-sdk-provider-zego/README.md", SourcePath: "providers/rtc-sdk-provider-zego/provider_package_contract.go", SourceSymbol: "RtcProviderZegoPackageContract", Builtin: false, RootPublic: false, Status: "future-runtime-bridge-only", RuntimeBridgeStatus: "reserved"},
    {ProviderKey: "livekit", PluginId: "rtc-livekit", DriverId: "sdkwork-rtc-driver-livekit", PackageIdentity: "github.com/sdkwork/rtc-sdk-provider-livekit", ManifestPath: "providers/rtc-sdk-provider-livekit/go.mod", ReadmePath: "providers/rtc-sdk-provider-livekit/README.md", SourcePath: "providers/rtc-sdk-provider-livekit/provider_package_contract.go", SourceSymbol: "RtcProviderLivekitPackageContract", Builtin: false, RootPublic: false, Status: "future-runtime-bridge-only", RuntimeBridgeStatus: "reserved"},
    {ProviderKey: "twilio", PluginId: "rtc-twilio", DriverId: "sdkwork-rtc-driver-twilio", PackageIdentity: "github.com/sdkwork/rtc-sdk-provider-twilio", ManifestPath: "providers/rtc-sdk-provider-twilio/go.mod", ReadmePath: "providers/rtc-sdk-provider-twilio/README.md", SourcePath: "providers/rtc-sdk-provider-twilio/provider_package_contract.go", SourceSymbol: "RtcProviderTwilioPackageContract", Builtin: false, RootPublic: false, Status: "future-runtime-bridge-only", RuntimeBridgeStatus: "reserved"},
    {ProviderKey: "jitsi", PluginId: "rtc-jitsi", DriverId: "sdkwork-rtc-driver-jitsi", PackageIdentity: "github.com/sdkwork/rtc-sdk-provider-jitsi", ManifestPath: "providers/rtc-sdk-provider-jitsi/go.mod", ReadmePath: "providers/rtc-sdk-provider-jitsi/README.md", SourcePath: "providers/rtc-sdk-provider-jitsi/provider_package_contract.go", SourceSymbol: "RtcProviderJitsiPackageContract", Builtin: false, RootPublic: false, Status: "future-runtime-bridge-only", RuntimeBridgeStatus: "reserved"},
    {ProviderKey: "janus", PluginId: "rtc-janus", DriverId: "sdkwork-rtc-driver-janus", PackageIdentity: "github.com/sdkwork/rtc-sdk-provider-janus", ManifestPath: "providers/rtc-sdk-provider-janus/go.mod", ReadmePath: "providers/rtc-sdk-provider-janus/README.md", SourcePath: "providers/rtc-sdk-provider-janus/provider_package_contract.go", SourceSymbol: "RtcProviderJanusPackageContract", Builtin: false, RootPublic: false, Status: "future-runtime-bridge-only", RuntimeBridgeStatus: "reserved"},
    {ProviderKey: "mediasoup", PluginId: "rtc-mediasoup", DriverId: "sdkwork-rtc-driver-mediasoup", PackageIdentity: "github.com/sdkwork/rtc-sdk-provider-mediasoup", ManifestPath: "providers/rtc-sdk-provider-mediasoup/go.mod", ReadmePath: "providers/rtc-sdk-provider-mediasoup/README.md", SourcePath: "providers/rtc-sdk-provider-mediasoup/provider_package_contract.go", SourceSymbol: "RtcProviderMediasoupPackageContract", Builtin: false, RootPublic: false, Status: "future-runtime-bridge-only", RuntimeBridgeStatus: "reserved"},
}

func GetRtcProviderPackageByProviderKey(providerKey string) *RtcProviderPackageCatalogEntry {
    for index := range OFFICIAL_RTC_PROVIDER_PACKAGES {
        if OFFICIAL_RTC_PROVIDER_PACKAGES[index].ProviderKey == providerKey {
            return &OFFICIAL_RTC_PROVIDER_PACKAGES[index]
        }
    }

    return nil
}

func GetRtcProviderPackageByPackageIdentity(packageIdentity string) *RtcProviderPackageCatalogEntry {
    for index := range OFFICIAL_RTC_PROVIDER_PACKAGES {
        if OFFICIAL_RTC_PROVIDER_PACKAGES[index].PackageIdentity == packageIdentity {
            return &OFFICIAL_RTC_PROVIDER_PACKAGES[index]
        }
    }

    return nil
}

File: sdkwork-rtc-sdk-rust/src/provider_package_catalog.rs
#[derive(Clone, Copy)]
#[allow(non_snake_case)]
pub struct RtcProviderPackageCatalogEntry {
    pub providerKey: &'static str,
    pub pluginId: &'static str,
    pub driverId: &'static str,
    pub packageIdentity: &'static str,
    pub manifestPath: &'static str,
    pub readmePath: &'static str,
    pub sourcePath: &'static str,
    pub sourceSymbol: &'static str,
    pub builtin: bool,
    pub rootPublic: bool,
    pub status: &'static str,
    pub runtimeBridgeStatus: &'static str,
}

pub struct RtcProviderPackageCatalog;

pub const OFFICIAL_RTC_PROVIDER_PACKAGES: [RtcProviderPackageCatalogEntry; 10] = [
    RtcProviderPackageCatalogEntry { providerKey: "volcengine", pluginId: "rtc-volcengine", driverId: "sdkwork-rtc-driver-volcengine", packageIdentity: "rtc-sdk-provider-volcengine", manifestPath: "providers/rtc-sdk-provider-volcengine/Cargo.toml", readmePath: "providers/rtc-sdk-provider-volcengine/README.md", sourcePath: "providers/rtc-sdk-provider-volcengine/src/lib.rs", sourceSymbol: "RtcProviderVolcenginePackageContract", builtin: true, rootPublic: false, status: "future-runtime-bridge-only", runtimeBridgeStatus: "reserved" },
    RtcProviderPackageCatalogEntry { providerKey: "aliyun", pluginId: "rtc-aliyun", driverId: "sdkwork-rtc-driver-aliyun", packageIdentity: "rtc-sdk-provider-aliyun", manifestPath: "providers/rtc-sdk-provider-aliyun/Cargo.toml", readmePath: "providers/rtc-sdk-provider-aliyun/README.md", sourcePath: "providers/rtc-sdk-provider-aliyun/src/lib.rs", sourceSymbol: "RtcProviderAliyunPackageContract", builtin: true, rootPublic: false, status: "future-runtime-bridge-only", runtimeBridgeStatus: "reserved" },
    RtcProviderPackageCatalogEntry { providerKey: "tencent", pluginId: "rtc-tencent", driverId: "sdkwork-rtc-driver-tencent", packageIdentity: "rtc-sdk-provider-tencent", manifestPath: "providers/rtc-sdk-provider-tencent/Cargo.toml", readmePath: "providers/rtc-sdk-provider-tencent/README.md", sourcePath: "providers/rtc-sdk-provider-tencent/src/lib.rs", sourceSymbol: "RtcProviderTencentPackageContract", builtin: true, rootPublic: false, status: "future-runtime-bridge-only", runtimeBridgeStatus: "reserved" },
    RtcProviderPackageCatalogEntry { providerKey: "agora", pluginId: "rtc-agora", driverId: "sdkwork-rtc-driver-agora", packageIdentity: "rtc-sdk-provider-agora", manifestPath: "providers/rtc-sdk-provider-agora/Cargo.toml", readmePath: "providers/rtc-sdk-provider-agora/README.md", sourcePath: "providers/rtc-sdk-provider-agora/src/lib.rs", sourceSymbol: "RtcProviderAgoraPackageContract", builtin: false, rootPublic: false, status: "future-runtime-bridge-only", runtimeBridgeStatus: "reserved" },
    RtcProviderPackageCatalogEntry { providerKey: "zego", pluginId: "rtc-zego", driverId: "sdkwork-rtc-driver-zego", packageIdentity: "rtc-sdk-provider-zego", manifestPath: "providers/rtc-sdk-provider-zego/Cargo.toml", readmePath: "providers/rtc-sdk-provider-zego/README.md", sourcePath: "providers/rtc-sdk-provider-zego/src/lib.rs", sourceSymbol: "RtcProviderZegoPackageContract", builtin: false, rootPublic: false, status: "future-runtime-bridge-only", runtimeBridgeStatus: "reserved" },
    RtcProviderPackageCatalogEntry { providerKey: "livekit", pluginId: "rtc-livekit", driverId: "sdkwork-rtc-driver-livekit", packageIdentity: "rtc-sdk-provider-livekit", manifestPath: "providers/rtc-sdk-provider-livekit/Cargo.toml", readmePath: "providers/rtc-sdk-provider-livekit/README.md", sourcePath: "providers/rtc-sdk-provider-livekit/src/lib.rs", sourceSymbol: "RtcProviderLivekitPackageContract", builtin: false, rootPublic: false, status: "future-runtime-bridge-only", runtimeBridgeStatus: "reserved" },
    RtcProviderPackageCatalogEntry { providerKey: "twilio", pluginId: "rtc-twilio", driverId: "sdkwork-rtc-driver-twilio", packageIdentity: "rtc-sdk-provider-twilio", manifestPath: "providers/rtc-sdk-provider-twilio/Cargo.toml", readmePath: "providers/rtc-sdk-provider-twilio/README.md", sourcePath: "providers/rtc-sdk-provider-twilio/src/lib.rs", sourceSymbol: "RtcProviderTwilioPackageContract", builtin: false, rootPublic: false, status: "future-runtime-bridge-only", runtimeBridgeStatus: "reserved" },
    RtcProviderPackageCatalogEntry { providerKey: "jitsi", pluginId: "rtc-jitsi", driverId: "sdkwork-rtc-driver-jitsi", packageIdentity: "rtc-sdk-provider-jitsi", manifestPath: "providers/rtc-sdk-provider-jitsi/Cargo.toml", readmePath: "providers/rtc-sdk-provider-jitsi/README.md", sourcePath: "providers/rtc-sdk-provider-jitsi/src/lib.rs", sourceSymbol: "RtcProviderJitsiPackageContract", builtin: false, rootPublic: false, status: "future-runtime-bridge-only", runtimeBridgeStatus: "reserved" },
    RtcProviderPackageCatalogEntry { providerKey: "janus", pluginId: "rtc-janus", driverId: "sdkwork-rtc-driver-janus", packageIdentity: "rtc-sdk-provider-janus", manifestPath: "providers/rtc-sdk-provider-janus/Cargo.toml", readmePath: "providers/rtc-sdk-provider-janus/README.md", sourcePath: "providers/rtc-sdk-provider-janus/src/lib.rs", sourceSymbol: "RtcProviderJanusPackageContract", builtin: false, rootPublic: false, status: "future-runtime-bridge-only", runtimeBridgeStatus: "reserved" },
    RtcProviderPackageCatalogEntry { providerKey: "mediasoup", pluginId: "rtc-mediasoup", driverId: "sdkwork-rtc-driver-mediasoup", packageIdentity: "rtc-sdk-provider-mediasoup", manifestPath: "providers/rtc-sdk-provider-mediasoup/Cargo.toml", readmePath: "providers/rtc-sdk-provider-mediasoup/README.md", sourcePath: "providers/rtc-sdk-provider-mediasoup/src/lib.rs", sourceSymbol: "RtcProviderMediasoupPackageContract", builtin: false, rootPublic: false, status: "future-runtime-bridge-only", runtimeBridgeStatus: "reserved" },
];

pub fn get_rtc_provider_package_by_provider_key(
    provider_key: &str,
) -> Option<&'static RtcProviderPackageCatalogEntry> {
    OFFICIAL_RTC_PROVIDER_PACKAGES
        .iter()
        .find(|entry| entry.providerKey == provider_key)
}
